sys/mips/mips/pm_machdep.c

```diff
@@ -310,13 +310,17 @@ sendsig(sig_t catcher, ksiginfo_t *ksi, sigset_t *mask)
 	regs->pc = (trapf_pc_t)catcher;
 	regs->c12 = regs->pc;
 	regs->csp = sfp;
-	regs->c17 = td->td_pcb->pcb_cherisignal.csig_sigcode;
+	regs->c17 = p->p_md.md_sigcode;
 
 	/*
 	 * For now only change IDC if we were sandboxed. This makes cap-table
 	 * binaries work as expected (since they need cgp to remain the same).
 	 *
 	 * TODO: remove csigp->csig_idc
+	 *
+	 * XXX: DDC used to be unconditional but was made conditional
+	 * as part of a commit to support legacy ABI.  Should it be
+	 * conditional on CHERIABI_LEGACY_SUPPORT?
 	 */
 	if (cheri_is_sandboxed) {
 		regs->ddc = td->td_pcb->pcb_cherisignal.csig_ddc;
```

Comment:
(Grumble, GH won't let you add comments on lines that haven't changed)

The line below this is the sole use of csig_idc, and csig_idc is always zero here.  Alex added the TODO about removing csig_idc in 1c3b97c9dac5a89358150e40b09d3b64486de35b.

This line and a similar line in hybridabi_sendsig() are the only uses of csig_ddc.  Currently csig_ddc is always the initial DDC (so NULL on purecap and all of userspace for hybrid), and we restore DDC to those values when running a signal handler.  For RISC-V and Morello we just use whatever DDC happens to be at the time of the signal.